### DISCUSSION TITLE
[firebase_database] The order in which you get the data

### DISCUSSION THREAD

**Discussion by TheHypnoo** [firebase/flutterfire#7637] [Q&A]
Hello, in the previous versions of firebase_database (less than 9.0.0) it returned the data ordered automatically, now it is messed up in its own way. The only solution is to do a sort? Because it happens now and before, each record was "created" in the database.

Thanks.

**Comment by deepak786** [comment]
From the version `9.0.0`, to get the ordered data you have to use `event.snapshot.children`
```
query.onValue.map((event) {
    List<T> list = [];
    // event.snapshot.children gives the ordered data
    for (var child in event.snapshot.children) {
      list.add(child.value); // here you need to convert child.value according to your requirement for e.g: T.fromJson(child.value)
    }
    //...
});
```

**Reply by TheHypnoo** [reply]
> From the version `9.0.0`, to get the ordered data you have to use `event.snapshot.children`
> 
> ```
> query.onValue.map((event) {
>     List<T> list = [];
>     // event.snapshot.children gives the ordered data
>     for (var child in event.snapshot.children) {
>       list.add(child.value); // here you need to convert child.value according to your requirement for e.g: T.fromJson(child.value)
>     }
>     //...
> });
> ```

@deepak786 

Right, I get it and it works. But the children are not ordered, only the root of that call.

Example:

```
/
 Users/ 
      uidUser/
         personal-information/
                 name: "Josh"
                 surname: "Smith"
                     phone/
                        t0: "+34 123456789"
                        t1: "+34 123456781"
```

Phone does not return it ordered, as would be the correct way to obtain it as it is in the ordered database.
